### DISCUSSION TITLE
"TypeError: Invalid value used as weak map key" when using helper function to create atoms.

### DISCUSSION THREAD

**Discussion by Noir-Lime** [pmndrs/jotai#2273] [Bug report]
## Summary
```
const suspendUntilSetAtom = function <T>() {
  return atom<T | Promise<T>>(new Promise<T>(() => {}));
};

const ATOM_select_records_request =
  suspendUntilSetAtom<I_SelectRecordsReq>();
const ATOM_I_Obj = suspendUntilSetAtom<I_Obj>();
```

```
const ATOM_select_records_request = atom<
  I_SelectRecordsReq | Promise<I_SelectRecordsReq>
>(new Promise<I_SelectRecordsReq>(() => {}));

const ATOM_I_Obj = atom<I_Obj | Promise<I_Obj>>(new Promise<I_Obj>(() => {}));
```

Ideally, these two implementations would have the same behavior, but the one above results in the following error:
![image](https://github.com/pmndrs/jotai/assets/133947209/9dc6174a-9981-4d5b-95dc-73c4dc4f36a7)

**Comment by dai-shi** [comment]
That seems like a weird error. Can you create a minimal reproduction with CodeSandbox?

**Reply by Noir-Lime** [reply]
Removing ```@swc-jotai/react-refresh``` solves the issue.

**Reply by Noir-Lime** [reply]
See this codesandbox: https://codesandbox.io/p/sandbox/runtime-wood-wlqq2m

**Comment by himself65** [comment]
Upstream issue: https://github.com/pmndrs/swc-jotai/issues/17
